# Problem 5

Show that any composite three-digit number must have a prime factor less than or equal to 31.

## Proof by Exhaustion

In [1]:
using Primes
P = Set(primes(31)) # P is the set of all primes less than or equal to 31
for n in 100:999 # all three-digit numbers
    isprime(n) && continue # we are only interested in composite numbers
    F = factor(Set, n) # F is the set of prime factors of n
    isempty(F ∩ P) && error("$n has no prime factor less than or equal to 31 [$(factor(n))]")
    # If we make it here, then n has at least one prime factor less than or equal to 31
end
println("Proved, by exhaustion.")

Proved, by exhaustion.


## Proof *via* Eratosthenes

From Eratosthenes we have that every composite number $a$ has a prime divisor $p$ satisfying $p \leq \sqrt{a}$. The largest three-digit number is $999$. $\lfloor \sqrt{999} \rfloor = 31$. So, $999$ and every composite number less than it has a prime factor less than or equal to $31$.

## Further Discussion

What’s the first number that doesn’t have a prime factor less than or equal to 31? Running the above code with a larger upper bound throws an error:

In [2]:
for n in 1000:9999 # all four-digit numbers
    isprime(n) && continue # we are only interested in composite numbers
    F = factor(Set, n) # F is the set of prime factors of n
    isempty(F ∩ P) && error("$n has no prime factor less than or equal to 31 [$(factor(n))]")
    # If we make it here, then n has at least one prime factor less than or equal to 31
end
println("Proved, by exhaustion.")

ErrorException: 1369 has no prime factor less than or equal to 31 [Primes.Factorization(37 => 2)]

So, the answer is $1369$ which is $37^2$. $37$ is the first prime greater than 31:

In [3]:
nextprime(1 + 31)

37

so, the smallest composite number without a prime factor less than or equal to 31 is the number with the *smallest/simplest* canonical form that includes just the next prime raised to a power greater than one.
\begin{equation}
n = \prod\limits_{i}{p_i^{k_i}} = 37^2 = 1369
\end{equation}